### DISCUSSION TITLE
nx buildable library

### DISCUSSION THREAD

**Discussion by Crisis2010** [ben-rogerson/twin.macro#397] [Q&A]
Hello! I try to build and publish library with nx workspace, but for some reason tw in not compiled while building and that cause RefferenceError
<img width="1094" alt="error" src="https://user-images.githubusercontent.com/24792724/115122594-5fd17880-9fc1-11eb-9a9d-24b29344e2c1.png">

**Comment by ben-rogerson** [comment]
Hey phil, check out the [2x open nx workspace pull requests](https://github.com/ben-rogerson/twin.examples/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc) for some ideas on how to fix this.

**Reply by ben-rogerson** [reply]
just saw you've found them already 👍

**Reply by Crisis2010** [reply]
Hi Ben! Thanks, already using it, but this problem actual on both of them.

You need just make any library buildable, and try to build it by `nx build lib1`, after that in dist folder tw(...) is still in UMD files, and because of that I got error

**Reply by Crisis2010** [reply]
I've tried everything, but so far no results. Seems like it need custom executor with changes in Babel with configured roll-up plugin. Still trying to figure it out...

**Reply by ben-rogerson** [reply]
ive seen that error come up when babel hasn't run over the code before it's used.  You'll need to make sure the babel config file is being picked up

**Reply by Crisis2010** [reply]
After many attempts, I finally found the problem. Babel picks up the configuration correctly, but for some reason, the NX Web Package builder, which uses rollup and rollup-plugin-typescript2 to build, breaks the further twin.macro replacement. I haven't been able to find a competent solution to the problem yet, but disabling:
`plugins = [
...
/* typescript({
        check: true,
        tsconfig: options.tsConfig,
        tsconfigOverride: {
          compilerOptions: {
            rootDir: options.entryRoot,
            allowJs: false,
            declaration: true,
            paths: compilerOptionPaths,
            target: config.format === 'esm' ? undefined : 'es5',
          },
        },
      }), */
...
]`
solves the problem, but removes typing .d.ts files from the build.

If needed, I can post this temporary fix until I find a way to fix the problem without disabling typing.
